app/src/main/java/org/zephyrsoft/trackworktime/EventListActivity.java

```diff
@@ -226,13 +226,20 @@ private static void insertSeparators(List<Event> eventList) {
 			Event cur = iter.next();
 			if (prev == null || !isOnSameDay(prev, cur)) {
 				iter.previous();
-				iter.add(new EventSeparator(WeekDayHelper.getWeekDayLongName(cur.getDateTime())));
+				String caption = toEventSeparatorCaption(cur.getDateTime());
+				iter.add(new EventSeparator(caption));
 				iter.next();
 			}
 			prev = cur;
 		}
 	}
 
+	private static String toEventSeparatorCaption(OffsetDateTime dateTime) {
+		String dayName = WeekDayHelper.getWeekDayLongName(dateTime);
+		String date = DateTimeUtil.formatLocalizedDate(dateTime.toLocalDate());
+		return dayName + ", " + date;
```

Comment:
Instead of formatting the date alone and then concatenating it with the week day name, it would also be possible to format the date together with the week day in the first place (by adding a new method, maybe `DateTimeUtil.formatLocalizedDateWithWeekDay(OffsetDateTime)`). It would make later localisation efforts a bit easier, but I'd merge this PR regardless. If you like to, you may change this.

Reply:
I saw it, but I was a bit lazy here. I've updated it, so it only formats once.